File: adrop-ads-example-ios-objective-c/ViewController.m
//
//  ViewController.m
//  adrop-ads-example-ios-objective-c
//
//  Created by Leo on 11/14/23.
//

#import "ViewController.h"
@import AdropAds;

@interface ViewController (AdropAds) <AdropBannerDelegate>

@end

@implementation ViewController (AdropAds)

- (void)onAdReceived:(AdropBanner * _Nonnull)banner {
    NSLog(@"onAdReceived");
}

- (void)onAdClicked:(AdropBanner * _Nonnull)banner {
    NSLog(@"onAdClicked");
}

- (void)onAdFailedToReceive:(AdropBanner * _Nonnull)banner :(enum AdropErrorCode)error {
    NSLog(@"onAdFailedToReceive");
}

@end


@interface ViewController ()
@property (weak, nonatomic) IBOutlet UIView *adContainerView;

@end

@implementation ViewController

- (void)viewDidLoad {
    [super viewDidLoad];
    
    AdropBanner* banner = [[AdropBanner alloc] initWithUnitId:@"ADROP_PUBLIC_TEST_UNIT_ID"];
    banner.delegate = self;
    [banner load];
    [_adContainerView addSubview:banner];
    banner.autoresizingMask = UIViewAutoresizingFlexibleWidth | UIViewAutoresizingFlexibleHeight;
    banner.translatesAutoresizingMaskIntoConstraints = false;
    banner.frame = _adContainerView.bounds;
}

@end

File: adrop-ads-example-ios/ViewController.swift
//
//  ViewController.swift
//  adrop-ads-example-ios
//
//  Created by Leo on 11/14/23.
//

import UIKit
import AdropAds

class ViewController: UIViewController {
    @IBOutlet weak var adContainer: UIView!
    
    override func viewDidLoad() {
        super.viewDidLoad()
        // Do any additional setup after loading the view.
        
        let bannerView = AdropBanner(unitId: "ADROP_PUBLIC_TEST_UNIT_ID")
        bannerView.delegate = self
        bannerView.load()
        adContainer.addSubview(bannerView)
        bannerView.autoresizingMask = [.flexibleWidth, .flexibleHeight]
        bannerView.translatesAutoresizingMaskIntoConstraints = false
        bannerView.frame = adContainer.bounds
    }
}

extension ViewController: AdropBannerDelegate {
    func onAdReceived(_ banner: AdropAds.AdropBanner) {
        print("onAdReceived")
    }
    
    func onAdClicked(_ banner: AdropAds.AdropBanner) {
        print("onAdClickced")
    }
    
    func onAdFailedToReceive(_ banner: AdropAds.AdropBanner, _ error: AdropAds.AdropErrorCode) {
        print("onAdFailedToReceive \(AdropErrorCodeToString(code: error))")
    }
}

